src/charm.py

```diff
@@ -115,31 +116,61 @@ def __init__(self, *args):
             self, jobs=[{"static_configs": [{"targets": [f"*:{PROMETHEUS_PORT}"]}]}]
         )
         self._logging = LogProxyConsumer(
-            self, relation_name="logging", log_files=LOG_PATHS, container_name="discourse"
+            self, relation_name="logging", log_files=LOG_PATHS, container_name=CONTAINER_NAME
         )
         self._grafana_dashboards = GrafanaDashboardProvider(self)
 
+    def _on_start(self, _: ops.StartEvent) -> None:
+        """Handle start event.
+
+        Args:
+            event: Event triggering the start event handler.
+        """
+        self._setup_and_activate()
+
+    def _on_upgrade_charm(self, _: ops.UpgradeCharmEvent) -> None:
+        """Handle upgrade charm event.
+
+        Args:
+            event: Event triggering the upgrade charm event handler.
+        """
+        self._setup_and_activate()
+
+    def _on_discourse_pebble_ready(self, _: ops.PebbleReadyEvent) -> None:
+        """Handle discourse pebble ready event.
+
+        Args:
+            event: Event triggering the discourse pebble ready event handler.
+        """
+        self._setup_and_activate()
+
+    def _redis_relation_changed(self, _: HookEvent) -> None:
+        """Handle redis relation changed event.
+
+        Args:
+            event: Event triggering the redis relation changed event handler.
+        """
+        self._setup_and_activate()
+
     def _on_database_created(self, _: DatabaseCreatedEvent) -> None:
         """Handle database created.
 
         Args:
             event: Event triggering the database created handler.
         """
-        if self.unit.is_leader():
```

Comment:
Out of curiosity, may I ask why the unit no longer needs to be the leader?

Reply:
It is actually checked down the line just before executing the migrations

Reply:
I re-added the missing check in execute_migrations